wordpress-mobile/WordPress-Android
# Issue/8319 copy comment url phase 1

Fixes #8319 

### **Description**
This PR introduces the ability for a user to copy the direct url of a comment to the clipboard - option number 2 from the issue ticket, including the "share" action on the snackbar. 

Note: This PR uses a FluxC commit hash and is does not include release notes yet

### **Action Bar Changes**
- More option added to action bar
- Edit & trash options have been relocated to the More menu
![more_only](https://user-images.githubusercontent.com/506707/80023980-61e72100-84ac-11ea-8455-fafc75ba85f8.png)

 
### **More Menu Notes**
- Edit and trash/delete functionality remain the same as when they were located on the action bar
- Copy link address option will not be available if the comment is in the Trashed or Spam status

![popmenu](https://user-images.githubusercontent.com/506707/80024028-70cdd380-84ac-11ea-9828-620640791cd4.png)
![more-when-trashed](https://user-images.githubusercontent.com/506707/80024036-73c8c400-84ac-11ea-9103-68d000d5073a.png)

### **Implementation Details**

- Updated comment_action_footer.xml
    - Removed the edit and trash buttons
    - Added more button
    - Added new vector drawable for the More icon
- Added a new menu resource called menu_comment_more.xml
- CommentDetailFragment
    - Added an onClickListener for the More button click to show the More popup menu
    - Moved logic to handle the edit and trash/delete buttons to the more menu handler
    - Prepare and show the more menu based upon comment status
    - Added logic to copy the url text to the clipboard
    - Added logic to show the snackbar and handle the Share action
- Added string resources as needed


### **Testing Instructions**

**Precursor**
    - A WP.com site that has multiple comments
    - A self-hosted site that has multiple comments. 
    
**Test More/Copy Link functionality (repeat below for each WP.com site and self-hosted site)**

- Log into WPAndroid
- Navigate to the site that has comments
- Tap Comments on the My Site tab to navigate to the comment list
- Select a comment from the comment list to open the comment detail view
- Tap the More button from the action bar to open the more menu
- Tap Copy link address 
    - The link has been copied to your clipboard. 
    - Scenario 1: Open an app in which you would type in text and use the paste function to copy the link address from the clipboard
    - Scenario 2: Tap Share from the snackbar and choose an app to share the copied link with

**Test Delete/Trash function (regression)**

- Log into WPAndroid
- Navigate to the site that has comments
- Tap Comments on the My Site tab to navigate to the comment list
- Select a comment from the comment list to open the comment detail view
- Tap the More button from the action bar to open the more menu
- Tap Trash
- The comment detail view closes and the user is redirected to the All comments list
- Navigate to the Trashed list using the drop down filter
- Tap on the comment you just trashed to open the comment detail view
- Tap on the More button 
    - The trash feature has been replaced with delete
    - The copy link address option is no longer visible


**PR submission checklist:**

- [X ] I have considered adding unit tests where possible.
- [X ] I have considered adding accessibility improvements for my changes.
- [X ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.
